presto-jdbc/src/main/java/com/facebook/presto/jdbc/AbstractConnectionProperty.java

```diff
@@ -216,6 +217,43 @@ private QueryInterceptor loadClass(String interceptor)
         }
     }
 
+    protected static final class HttpProtocolConverter
+            implements Converter<List<Protocol>>
+    {
+        public static final HttpProtocolConverter HTTP_PROTOCOL_CONVERTER = new HttpProtocolConverter();
+        private HttpProtocolConverter() {}
+
+        @Override
+        public List<Protocol> convert(String value)
+        {
+            return Splitter.on(',').splitToList(value).stream()
+                    .map(this::loadProtocol)
+                    .distinct()
+                    .collect(toImmutableList());
+        }
+
+        private Protocol loadProtocol(String protocolName)
+        {
+            try {
+                switch (protocolName.toLowerCase(ENGLISH)) {
+                    case "http11":
+                        return Protocol.HTTP_1_1;
+                    case "http10":
+                        return Protocol.HTTP_1_0;
+                    case "http2":
+                        return Protocol.HTTP_2;
+                    case "spdy":
+                        return Protocol.SPDY_3;
```

Comment:
I just looked at `OkHttpClient.Builder#protocols` and noticed this line:

```
...
// Remove protocols that we no longer support.
protocols.remove(Protocol.SPDY_3);
...
```

Based on that, I don't think it's worthwhile to make it impossible to pass- but do suggest that we don't give SPDY_3 a "friendly name".

Reply:
Removed the friendly `"spdy"` option for protocols in latest push.